app/models/asset.rb

```diff
@@ -230,6 +230,23 @@ def post_process_file(team = nil)
       # Update asset's estimated size immediately
       update_estimated_size(team)
     end
+
+    if is_image?
+      Asset.delay(queue: :assets).calculate_quality(id) if file.content_type == 'image/jpeg'
+    end
+  end
+
+  def self.calculate_quality(id)
+    asset = find_by_id(id)
+    return unless asset.present? && asset.file.present?
```

Comment:
- file? and present? checks if the _file_name field is populated
 - exists? checks if the file exists (will perform a TCP connection if stored in the cloud)